include/grpc/grpc_security.h

```diff
@@ -1117,17 +1117,17 @@ grpc_authorization_policy_provider_static_data_create(
 GRPCAPI void grpc_authorization_policy_provider_release(
     grpc_authorization_policy_provider* provider);
 
-struct grpc_status_or_bool{
-  grpc_status_code status;
-  bool boolean;
-};
+//struct grpc_status_or_bool{
+//  grpc_status_code status;
+//  bool boolean;
+//};
 
 /**
  * Checks if the private key matches the certificate's public key.
  * Returns a not-OK status on failure, or a bool indicating
  * whether the key/cert pair matches.
  */
-GRPCAPI grpc_status_or_bool* grpc_tls_certificate_key_match(
+GRPCAPI grpc_status_code* grpc_tls_certificate_key_match(
     const char* private_key, const char* cert_chain);
```

Comment:
Hey, why do we need `grpc_tls_certificate_key_match`?

Reply:
I am trying to expose this function to the c-core. I am waiting on Zhen's feedback on this

Reply:
I get what you are trying to do. What I don't understand is why.

Reply:
Why expose it? Well as we noted earlier, the certificate-key match functionality could be highly useful in other situations/classes (not just the FileWatcherCertificateProvider). This should allow for it to be used where needed. I am not sure exactly if Zhen already has a use in mind, but I can recall Ashitha mentioning a few spots where she actually wanted me to implement it.

Reply:
Sounds good.